### ISSUE TITLE
Move HTML unit depedency to bom/application/pom.xml 

### ISSUE THREAD

**Issue by sberyozkin** [quarkusio/quarkus#28452]
### Description

`net.sourceforge.htmlunit:htmlunit` is currently located in `build-parent.pom` [here](https://github.com/quarkusio/quarkus/blob/main/build-parent/pom.xml#L292).
I'd like to propose to move it to `bom/application/pom.xml` as `quickstarts/security-openid-connect-web-authentication` and `quickstarts/security-openid-connect-web-authentication` use it for testing and it is also recommended to use it for testing OIDC code flow in the OIDC docs, so it would be good if users did not have to enter specific versions and exclusions.

It is really an equivalent to RestAssured (which is in the application bom) but used mainly for testing interactive endpoints.

@aloubyansky @gsmet Are you OK with this proposal ?

**Comment by aloubyansky** [comment]
Do we need a specific version of `net.sourceforge.htmlunit:htmlunit` or is it mostly for convenience? Also are the exclusions required in the applications for this dependency?
If there is no specific version constraint we need to be enforcing on the applications, I think we shouldn't be adding it.

**Comment by sberyozkin** [comment]
>Do we need a specific version of net.sourceforge.htmlunit:htmlunit or is it mostly for convenience?

For convenience, like for for other test dependencies we have in `bom/application/pom.xml`

>Also are the exclusions required in the applications for this dependency?

I don't know why these exclusions have been added 

>If there is no specific version constraint we need to be enforcing on the applications, I think we shouldn't be adding it.

I'm not sure I follow. For example, I have realized this fragment does not work: https://quarkus.io/guides/security-openid-connect-web-authentication#integration-testing - I copied it from one of the integration tests where it does work because we have the version in `build-parent/pom` and then I've just opened [this PR](https://github.com/quarkusio/quarkus/pull/28327/files#diff-95192f8049e4b0a3a326ebbd353c3428b121c1088f3aff807a0a8dee3d58a264R415) and it also does not work.

I.e, if users want to tests REST endpoints which return some data, we have `rest-assured` in `bom/application/pom.xml` - but `restassured` is not suitable if the rest endpoint returns a form which needs to be submitted back, so `htmlunit` is there for it, so why can't we make it convenient for users as well ?

**Comment by gsmet** [comment]
Can't say I'm excited either. Can you check if they have frequent CVEs?

**Comment by sberyozkin** [comment]
I'm puzzled as to what the actual problem is.

https://www.cvedetails.com/product/92763/Htmlunit-Project-Htmlunit.html?vendor_id=24365.

Can we have a test scope set in the dependencyManagement section ? It is only meant to be a test scoped dependency

**Comment by aloubyansky** [comment]
To me it sounds more like a guide maintenance related issue so far. That alone shouldn't be the reason to add version constraints to the BOM. I am trying to see whether there is more to it.
RestAssured, imo, is different in that we pretty much made it our official testing API and removing it will simply break a lot of projects (which is one of the reasons removal of assertj was also taking a long time).

> > Do we need a specific version of net.sourceforge.htmlunit:htmlunit or is it mostly for convenience?
> 
> For convenience, like for for other test dependencies we have in `bom/application/pom.xml`

Just for the info, we used to have more popular test deps in the quarkus-bom, such as assertj, which was later removed https://github.com/quarkusio/quarkus/pull/18006. At some point we moved a few other test related deps out the quarkus-bom. Sometimes users prefer different versions of those, some times they actually caused integration related issues (including among platform members).

> Can we have a test scope set in the dependencyManagement section ? It is only meant to be a test scoped dependency

Scope is a different concern. In general, it's better not to specify scopes in the BOM. Because then, in addition to the version, the scope will also be enforced unless explicitly overridden in user POMs. That could lead to issues that would take time to figure out.
